### PR TITLE
[17.0][MIG] mrp_bom_tracking: Migration to 17.0

### PR DESCRIPTION
**Pull request by AndreuOForgeFlow**
Standard migration

@ForgeFlow

### CODE REVIEW

File: mrp_bom_tracking/models/mrp_bom.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                    product_id = self.env["product.product"].browse(product_id)
+                product_id = product_id or lines.product_id
+                if lines:
+                    message_body = template_env._render_template(
+                        "mrp_bom_tracking.track_bom_template_2",
+                        values={"lines": lines, "product_id": product_id},
+                    )
+                    bom.message_post(
+                        body=message_body, subtype_id=self.env.ref("mail.mt_note").id
+                    )
```
Comment (GuillemCForgeFlow) [inline mrp_bom_tracking/models/mrp_bom.py]:
```suggestion
        if "product_id" in values:
            for bom_line in self:
                bom = bom_line.bom_id
                product_id = values.get("product_id")
                product = self.env["product.product"].browse(product_id)
                if product.exists():
                    message_body = template_env._render_template(
                        "mrp_bom_tracking.track_bom_template_2",
                        values={"lines": bom_line, "product_id": product},
                    )
                    bom.message_post(
                        body=message_body, subtype_id=self.env.ref("mail.mt_note").id
                    )
```
`product_id` check was already done at the beginning of the indentation, no need to do it again. Also, refactor to iterate over self, which contains the BoM lines, as before the change, it was already doing one message posting per line change. I believe this refactoring simplifies the understanding of the logic. What do you think?
I believe this is more of a general improvement, nothing related to the migration itself.

File: mrp_bom_tracking/models/mrp_bom.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+                        "mrp_bom_tracking.track_bom_line_template",
+                        values={
+                            "lines": lines,
+                            "product_qty": product_qty,
+                            "product_uom_id": product_uom_id,
+                        },
+                    )
+                    bom.message_post(
+                        body=message_body, subtype_id=self.env.ref("mail.mt_note").id
+                    )
```
Comment (GuillemCForgeFlow) [inline mrp_bom_tracking/models/mrp_bom.py]:
```suggestion
        elif "product_qty" in values or "product_uom_id" in values:
            for bom_line in self:
                bom = bom_line.bom_id
                product_qty = values.get("product_qty") or bom_line.product_qty
                product_uom_id = values.get("product_uom_id")
                if product_uom_id:
                    product_uom = self.env["uom.uom"].browse(product_uom_id)
                product_uom = product_uom or bom_line.product_uom_id
                message_body = template_env._render_template(
                    "mrp_bom_tracking.track_bom_line_template",
                    values={
                        "lines": bom_line,
                        "product_qty": product_qty,
                        "product_uom_id": product_uom,
                    },
                )
                bom.message_post(
                    body=message_body, subtype_id=self.env.ref("mail.mt_note").id
                )
```
Same as above comment.

File: mrp_bom_tracking/models/mrp_bom.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 ForgeFlow S.L. (https://www.forgeflow.com)
```
Comment (GuillemCForgeFlow) [inline mrp_bom_tracking/models/mrp_bom.py:1]:
```suggestion
# Copyright 2019-2024 ForgeFlow S.L. (https://www.forgeflow.com)
```
update Copyright here as well!

File: mrp_bom_tracking/tests/test_mrp_bom_tracking.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+# Copyright 2019 ForgeFlow S.L. (https://www.forgeflow.com)
```
Comment (GuillemCForgeFlow) [inline mrp_bom_tracking/tests/test_mrp_bom_tracking.py:1]:
```suggestion
# Copyright 2019-2024 ForgeFlow S.L. (https://www.forgeflow.com)
```
same here.

File: mrp_bom_tracking/views/bom_template.xml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- Copyright 2019 ForgeFlow S.L. (https://www.forgeflow.com)
```
Comment (GuillemCForgeFlow) [inline mrp_bom_tracking/views/bom_template.xml:2]:
```suggestion
<!-- Copyright 2019-2024 ForgeFlow S.L. (https://www.forgeflow.com)
```
and here.

**Review by GuillemCForgeFlow — CHANGES_REQUESTED** [review]
general improvements and feedback 🙂

File: mrp_bom_tracking/models/mrp_bom.py (hunk 3 above)
Comment (AndreuOForgeFlow) [reply mrp_bom_tracking/models/mrp_bom.py:1]: It is not necessary to do that if we want to follow the OCA migration guidelines to v17 as seen here: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0#tasks-not-to-do-in-the-migration

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM! Thank you for the improvements 👍🏽

**Review by LoisRForgeFlow — COMMENTED** [review]
Tested and working good, I want to suggest one more improvement: Can you add tracking to the fields in the Miscelaneous tab too?

image

Thanks!

**Comment by AndreuOForgeFlow** [comment]
> Tested and working good, I want to suggest one more improvement: Can you add tracking to the fields in the Miscelaneous tab too?
> 
> image
> 
> Thanks!

Done @LoisRForgeFlow!

**Review by LoisRForgeFlow — APPROVED** [review]
Thanks!

/ocabot merge nobump

**Pull request merged**